encoding/descaling.md

```diff
@@ -0,0 +1,452 @@
+I'll be using `split` and `join` from `kagefunc` to split the planes
+and then join them again in this example,
+and `get_w` from `vsutil` to calculate the width)[^1]:
+
+[Comparison between frames][manaria_compare]
+
+```Py
+import vapoursynth as vs
+import vsutil
+import kagefunc as kgf
+import fvsfunc as fvf
+core = vs.core
+
+src = core.lsmas.LWLibavSource("BDMV/[BDMV][190302][マナリアフレンズ I]/BD/BDMV/STREAM/00007.m2ts")
+src = fvf.Depth(src, 32)
+
+Y, U, V = kgf.split(src)
+height = 878
+width = vsutil.get_w(height)
+
+descale_a = core.descale.Debilinear(Y, width, height).resize.Bilinear(1920, 1080)                                                        # Bilinear
+descale_b = core.descale.Debicubic(Y, width, height, b=1/3, c=1/3).resize.Bicubic(1920, 1080, filter_param_a=1/3, filter_param_b=1/3)    # Mitchell-Netravali
+descale_c = core.descale.Debicubic(Y, width, height, b=0, c=1).resize.Bicubic(1920, 1080, filter_param_a=0, filter_param_b=1)            # Sharp Bicubic
+descale_d = core.descale.Debicubic(Y, width, height, b=1, c=0).resize.Bicubic(1920, 1080, filter_param_a=1, filter_param_b=0)            # B-Spline
+descale_e = core.descale.Debicubic(Y, width, height, b=0, c=1/2).resize.Bicubic(1920, 1080, filter_param_a=0, filter_param_b=1/2)        # Catmull-rom
+descale_f = core.descale.Despline36(Y, width, height).resize.Spline36(1920, 1080)                                                        # Spline36
+
+descaled = kgf.join([descale_a, U, V])
+descaled.set_output()
+```
+
+You might realize that after descaling,
+we are immediately upscaling the frame
+with the same kernel and values again.
+This is done so we can compare the before and after.
+The closer the new frame is to the old one,
+how more likely it is that you've got the correct kernel.
+Zooming in on the frame at 4x magnification or higher
+will help immensely.
+An alternative that you can do
+is to simply descale until
+you've got what you believe to be the best result.
+It's faster to do it this way,
+but might be less accurate.
+
+[manaria_compare]: https://slowpics.org/comparison/61e39e1e-d074-4d83-b7c9-b0f4e1861855
+
+
+## Credits and other native 1080p elements
+
+There is one very, very important thing
+to keep in mind when descaling:
+
+*Credits are usually done in 1080p*.
+
+There are various masks you can use
+to help with dealing with that issue,
+but it might be better
+to make use of existing wrappers instead.
+For this example I'll
+be using `inverse_scale` from kagefunc.
+
+```Py
+descaled = kgf.inverse_scale(src, height=878, kernel='bicubic', b=0, c=1/2, mask_detail=True)
+```
+
+We can make use of the mask
+that `inverse_scale` uses internally
+as well.
+
+```Py
+descaled = kgf.inverse_scale(src, height=874, kernel='bicubic', b=0, c=1/2)
+descaled_mask = kgf._generate_descale_mask(vsutil.get_y(core.resize.Spline36(src, descaled.width, descaled.height)), vsutil.get_y(descaled), kernel='bicubic', b=0, c=1/2)
```

Comment:
You’re not supposed to call methods prefixed with `_`. Use `mask_descale` instead.

Reply:
For this example I wanted to generate the mask specifically. As far as I can tell, I can't do that with `mask_descale`.

Reply:
Nvm, I’m an idiot. I thought I put a debug parameter in there.